File: En Kotlin - JVM/Palindromos.kt
/*
*    Autor:  Fixed Torres
*    Descp:  Evaluar si una palabra o frase es o no es palindromo.
*/

package palindromos

fun main(Args: Array<String>) {
    var (igual, aux) = Pair(0, 0)
    var texto: String

    print("Ingrese la palabra que desea evaluar: ")
    texto = readLine() as String

    for (ind in texto.length - 1 downTo 0) {
        when { texto[ind] == texto[aux] -> igual++ }
        aux++
    }

    when (igual) {
        texto.length -> println("$texto es palindromo :D")
        else -> println("$texto no es palindromo :(")
    }
}

File: En Ruby - Interpretado/Palindromos.rb
print "Ingrese la palabra que desea evaluar: "
texto = gets().chomp("\n")

igual, aux = 0, texto.length - 1

0.step texto.length - 1 do |n|
  if texto[n].downcase == texto[aux].downcase
    igual += 1
  end
  aux -= 1
end

if texto.length == igual
  puts "La palabra ingresada si es palíndromo"
else
  puts "La palabra ingresada no es palíndromo"
end
